cpr/threadpool.cpp

```diff
@@ -0,0 +1,144 @@
+#include "cpr/threadpool.h"
+
+namespace cpr {
+
+ThreadPool::ThreadPool(int min_threads, int max_threads, int max_idle_ms)
+    : min_thread_num(min_threads)
+    , max_thread_num(max_threads)
+    , max_idle_time(max_idle_ms)
+    , status(STOP)
+    , cur_thread_num(0)
+    , idle_thread_num(0)
+{}
+
+ThreadPool::~ThreadPool() {
+    stop();
+}
+
+int ThreadPool::start(int start_threads) {
```

Comment:
Based on the cpr convention all method names should start with an uppercase letter.